### PR TITLE
only limit types in stack traces in the REPL

### LINKED ISSUE #50575: regression in stacktrace correctness with type information truncation
```
 [23] include_package_for_output(pkg::Base.PkgId, input::String, depot_path::Vector{…}, dl_load_path::Vector{…}, load_path::Vector{…}, concrete_deps::Vector{…}, source::String)
    @ Base ./loading.jl:2216
 [24] top-level scope
    @ stdin:3
Some type information was truncated. Use `show(err)` to see complete types.
in expression starting at /home/pkgeval/.julia/packages/SimpleWorkflows/GNXa7/src/SimpleWorkflows.jl:1
in expression starting at stdin:3
```
https://s3.amazonaws.com/julialang-reports/nanosoldier/pkgeval/by_date/2023-07/14/SimpleWorkflows.primary.log

This now says "use `show(err)`", but the `err` variable does not exist, so this message is now factually incorrect.

### PR DESCRIPTION
**Pull request by JeffBezanson**
fixes #50575

### CODE REVIEW

**Comment by vtjnash** [comment]
There is also the issue that the stacktraces are getting truncated incorrectly on CI, so this does not close https://github.com/JuliaLang/julia/issues/50575 as that is still a regression

**Comment by JeffBezanson** [comment]
OK I guess a good way to handle it is to introduce a new IO context flag for limiting stack trace types and only set it from the REPL?

**Comment by vtjnash** [comment]
meaning the `stacktrace_types_limited` flag perhaps?

**Comment by JeffBezanson** [comment]
Yes we could reuse that.

File: base/show.jl [hunk 1]
```diff
@@ -2559,14 +2559,12 @@ function show_tuple_as_call(out::IO, name::Symbol, sig::Type;
     print_within_stacktrace(io, ")", bold=true)
     show_method_params(io, tv)
     str = String(take!(unwrapcontext(io)[1]))
-    if get(out, :limit, false)::Bool
+    typelimitflag = get(out, :stacktrace_types_limited, nothing)
+    if typelimitflag !== nothing
```
Comment (aviatesk) [inline base/show.jl:2563]:
```suggestion
    if typelimitflag isa RefValue{Bool}
```
for better inference?

**Pull request merged**